### PR TITLE
Closes #521: Bump dev dependency version numbers and refactor Actions.

### LINKED ISSUE #521: Dev dependencies need routine updates (early August, 2022).
## Conditions of satisfaction
- [ ] Dependabot should show fewer proposed updates.
- [ ] The dev dependencies should be similar to upstream twbs/bootstrap v4-dev and main branches.

### PR DESCRIPTION
**Pull request by mmunro-ltrr**
This includes yet another fragile refactoring of the GitHub actions setup, so might need more work before actual review. Some dependabot-suggested version number bumps do not appear, because twbs/bootstrap has ignored them for now.

### CODE REVIEW

**Comment by trackleft** [comment]
The changes here are probably ok https://github.com/az-digital/arizona-bootstrap/pull/522/files#diff-1a9dd1a33d61bc61437d2aa16f3d01169a25cb21fa3b8e395d992a1d264cdcca

@akslay @danahertzberg

File: dist/css/arizona-bootstrap.css [hunk 1]
```diff
@@ -6204,7 +6224,7 @@ a.close.disabled {
   width: 2rem;
   height: 2rem;
   vertical-align: -0.125em;
-  border: 0.25em solid currentColor;
+  border: 0.25em solid currentcolor;
```
Comment (trackleft) [inline dist/css/arizona-bootstrap.css:6227]: Never heard of this but, apparently it is a thing https://css-tricks.com/currentcolor/

**Comment by mmunro-ltrr** [comment]
Unfortunately the review site is almost completely broken (which prompted https://github.com/az-digital/arizona-bootstrap/issues/523). I didn't want to relegate this PR to a WIP, though (wanting to test any updates as changes to an active PR). The breakage has no connection with currentcolor, but the baseURL not getting included in many of the generated links.

**Comment by mmunro-ltrr** [comment]
This should really be ready to merge, after the latest changes. Invalid relative URLs caused the broken review site; they had been present all along, but a bug in Hugo was letting them work: when this was fixed in the v101 release, the existing problem started manifesting itself. The most recent commit with version number bumps actually changes the included version of upstream twbs Bootstrap to 4.6.2, with the main consequence being extra noise in the diffs (from updated version number comments).

A big hope of this PR is that it will finally close https://github.com/az-digital/arizona-bootstrap/issues/420 and obsolete the draft https://github.com/az-digital/arizona-bootstrap/pull/431 allowing Arizona Bootstrap to use current versions of Hugo to build the documentation files that are also used in some of the linting checks. Although GitHub Actions were using a Docker image built from the same Dockerfile used in local dev environments, the default user in the container wasn't the `node` user defined in the image, but one that was a default in the GitHub Action runner. A routine update to Hugo changed one of its internal dependencies to a version that tried to access the home directory of the user under which it was running, but no such home directory existed within the container. Forcing more operations within the container to run as the `node` user, and fixing the file ownership to prevent this conflicting with the default GitHub Action user seems to be working, but a follow-up check will be that local dev environments are not affected (my local setup has the default user outside the container using the same uid as the `node` user inside, so is a bad test case).

**Review by joeparsons — APPROVED** [review]

File: dist/css/arizona-bootstrap.css (hunk 1 above)
Comment (bberndt-uaz) [reply dist/css/arizona-bootstrap.css:6227]: Yes, it looks like `currentcolor` is correct: https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#currentcolor_keyword

**Review by trackleft — APPROVED** [review]

**Pull request merged**